shield.go

```diff
@@ -17,6 +18,8 @@ const MaxLength = 4096
 // SaltLength is the length of the salt generated by GenerateSalt.
 const SaltLength = 32
 
+var ErrPasswordTooLong = errors.New("password too long")
```

Comment:
Thanks for the explanation! In that case I think I prefer your solution instead of the type definition.